repository/src/main/java/com/epam/esm/repository/impl/CertificateRepositoryImpl.java

```diff
@@ -0,0 +1,129 @@
+package com.epam.esm.repository.impl;
+
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.exception.DataException;
+import com.epam.esm.repository.SearchCriteria;
+import com.epam.esm.repository.builder.QueryBuilder;
+import com.epam.esm.repository.mapper.CertificateMapper;
+import com.epam.esm.repository.repositoryinterfaces.CertificateRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static com.epam.esm.repository.Parameters.*;
+import static com.epam.esm.repository.builder.CertificateQueries.*;
+
+@Repository
+public class CertificateRepositoryImpl implements CertificateRepository {
+
+    private final CertificateMapper rowMapper;
+    private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+    private final QueryBuilder queryBuilder;
+
+    @Autowired
+    public CertificateRepositoryImpl(CertificateMapper rowMapper,
+                                     DataSource dataSource,
+                                     QueryBuilder queryBuilder) {
+        this.rowMapper = rowMapper;
+        this.namedParameterJdbcTemplate = new NamedParameterJdbcTemplate(dataSource);

```

Comment:
Please do not create JdbcTemplate in constructor. Declare it in @Configuration class and inject as complete bean

Reply:
fixed